x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/action/GetSearchApplicationAction.java

```diff
@@ -99,14 +128,46 @@ public Response(
             long updatedAtMillis,
             SearchApplicationTemplate template
         ) {
+            if (Strings.isNullOrEmpty(name)) {
+                throw new IllegalArgumentException("Search Application name cannot be null or blank");
+            }
+            // if (name!=null){
             this.searchApp = new SearchApplication(name, indices, analyticsCollectionName, updatedAtMillis, template);
+            // }
+            // else {
+            // this
+            // }
```

Comment:
Let's remove commented code:

```suggestion
            this.searchApp = new SearchApplication(name, indices, analyticsCollectionName, updatedAtMillis, template);
```